i dont actually understand what do you mean by this ,if you are refering to assert condtion the test will fail
```
In [92]: Ap, rep = posify(A)

In [93]: assert Ap == A.subs(rep)
```
```
        A = Matrix([[1, 2, 3], [4, 5, 6 * Abs(x)]])
        Ap, rep = posify(A)
>       assert Ap == A.subs(rep)
E       assert Matrix([\n[1, ...[4, 5, 6*_x]]) == Matrix([\n[1, ...5, 6*Abs(x)]])
E
E         Use -v to get more diff

test_simplify.py:553: AssertionError
                                                          DO *NOT* COMMIT!
===================================================== short test summary info ======================================================
FAILED test_simplify.py::test_posify - assert Matrix([\n[1, ...[4, 5, 6*_x]]) == Matrix([\n[1, ...5, 6*Abs(x)]])
======================================================== 1 failed in 6.70s =========================================================
```
Looking forward for response.

Thanks

--- a/sympy/simplify/simplify.py
+++ b/sympy/simplify/simplify.py
@@ -264,7 +264,7 @@ def posify(eq):
     [2]
     """
     eq = sympify(eq)
-    if iterable(eq):
+    if not isinstance(eq, Basic) and iterable(eq):
         f = type(eq)
         eq = list(eq)
         syms = set()

--- a/sympy/simplify/tests/test_simplify.py
+++ b/sympy/simplify/tests/test_simplify.py
@@ -548,6 +548,10 @@ def test_posify():
     assert str(Sum(posify(1/x**n)[0], (n,1,3)).expand()) == \
         'Sum(_x**(-n), (n, 1, 3))'
 
+    A = Matrix([[1, 2, 3], [4, 5, 6 * Abs(x)]])
+    Ap, rep = posify(A)
+    assert Ap == A.subs(*reversed(rep.popitem()))
+
     # issue 16438
     k = Symbol('k', finite=True)
     eq, rep = posify(k)

